### PR TITLE
Tests for current VMI tools

### PR DESCRIPTION
**Pull request by MikhailRyazanov**
Unit tests for recording the current behavior of `abel.tools.vmi`, such that future corrections (see #312) can maintain backward compatibility or detect where and how it is broken.

The file `abel/tests/test_tools.py`, which contained a mix of very superficial tests of VMI and centering tools, is deleted: the imporoved VMI tests are moved to `abel/tests/test_tools_vmi.py`, and these centering tests, already covered by `abel/tests/test_tools_center.py`, are simply removed.

### CODE REVIEW

**Comment by MikhailRyazanov** [comment]
@DanHickstein and @stggh, something strange happens with GitHub (or on my side): I cannot add any reviewers (neither labels, although can assign myself). Could you please add yourself or each other as a reviewer to this PR? Just to check.

**Comment by DanHickstein** [comment]
Huh, you have "Admin" access to the repository, so that should let you do anything you want on the webpage as far as I understand.

**Comment by MikhailRyazanov** [comment]
I know, and it all worked fine before... Perhaps, GitHub again broke something in “old” (~1 year old) browsers. :–/

**Comment by DanHickstein** [comment]
You're going to have to upgrade from Netscape Navigator sometime... :)

**Comment by MikhailRyazanov** [comment]
If nobody objects, I'm going to merge this and move on to adding the new functions...

**Review by DanHickstein — APPROVED** [review]

**Comment by DanHickstein** [comment]
Yes, this seems fine to me. Even if not yet perfect, seems better than what was there before. Fine with me to merge.

File: abel/tests/test_tools_vmi.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                    err_msg='Amplitude')
+
+
+def test_toPES():
+    """
+    Check that toPES at least does not crash.
+    TODO: do some meaningful tests.
+    """
+    # test image (not projected)
+    IM = SampleImage(name='dribinski').image
```
Comment (stggh) [inline abel/tests/test_tools_vmi.py:132]: `IM = SampleImage(name='Ominus').image`, the pseudo O<sup>-</sup> photoelectron spectrum would be more realistic for this test, and the following parameters give something close to reality, with the dominant peak positioned around 11784.68 cm<sup>-1</sup>, the electron affinity of oxygen:
```python
    # test image (not projected)
    IM = SampleImage(name='Ominus').image

    # (parameters from example_hansenlaw.py, although for a different image)
    eBE, PES = toPES(*angular_integration(IM),
                     energy_cal_factor=1.2e-5,
                     photon_energy=1.0e7/808.6, Vrep=-100,
                     zoom=IM.shape[-1]/2048)

    assert_allclose(eBE[PES.argmax()], 11784.68, 0.1)

```
Screenshot-20210428084442-588x417

`n=1001` would be better for realism, but not essential for this test.

**Review by stggh — APPROVED** [review]

**Comment by stggh** [comment]
My python 3.8.6 claims `"""` is depreciated?  (not related to this PR):
```python
[Gibbo:.+el-MR-tools.vmi]$ pytest-3 
============================= test session starts ==============================
platform linux -- Python 3.8.6, pytest-4.6.11, py-1.9.0, pluggy-0.13.0
rootdir: /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi
plugins: cov-2.11.1
collected 90 items                                                             

abel/tests/test_basex.py .............                                   [ 14%]
abel/tests/test_benchmarks.py ..                                         [ 16%]
abel/tests/test_dasch_methods.py .....                                   [ 22%]
abel/tests/test_direct.py .....                                          [ 27%]
abel/tests/test_hansenlaw.py .......                                     [ 35%]
abel/tests/test_linbasex.py ..                                           [ 37%]
abel/tests/test_math.py .                                                [ 38%]
abel/tests/test_onion_bordas.py ......                                   [ 45%]
abel/tests/test_rbasex.py ...............                                [ 62%]
abel/tests/test_tools_center.py ......                                   [ 68%]
abel/tests/test_tools_circularize.py .                                   [ 70%]
abel/tests/test_tools_distributions.py ................                  [ 87%]
abel/tests/test_tools_polynomial.py .....                                [ 93%]
abel/tests/test_tools_symmetry.py .                                      [ 94%]
abel/tests/test_tools_vmi.py .....                                       [100%]

=============================== warnings summary ===============================
abel/__init__.py:23
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/__init__.py:23: DeprecationWarning: invalid escape sequence \.
    filterwarnings('default', '^abel\.', category=DeprecationWarning)

abel/tools/analytical.py:164
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/analytical.py:164: DeprecationWarning: invalid escape sequence \ 
    """

abel/tools/analytical.py:222
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/analytical.py:222: DeprecationWarning: invalid escape sequence \ 
    """

abel/tools/math.py:64
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/math.py:64: DeprecationWarning: invalid escape sequence \m
    """

abel/tools/polynomial.py:18
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/polynomial.py:18: DeprecationWarning: invalid escape sequence \ 
    """

abel/tools/polynomial.py:193
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/polynomial.py:193: DeprecationWarning: invalid escape sequence \ 
    """

abel/tools/transform_pairs.py:56
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/transform_pairs.py:56: DeprecationWarning: invalid escape sequence \s
    """ coefficient

abel/tools/transform_pairs.py:66
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/transform_pairs.py:66: DeprecationWarning: invalid escape sequence \e
    """**profile1**:

abel/tools/transform_pairs.py:149
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/transform_pairs.py:149: DeprecationWarning: invalid escape sequence \e
    """**profile2**:

abel/tools/transform_pairs.py:198
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/transform_pairs.py:198: DeprecationWarning: invalid escape sequence \e
    """**profile3**:

abel/tools/transform_pairs.py:268
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/transform_pairs.py:268: DeprecationWarning: invalid escape sequence \ 
    """**profile4**:

abel/tools/transform_pairs.py:376
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/transform_pairs.py:376: DeprecationWarning: invalid escape sequence \e
    """**profile5**:

abel/tools/transform_pairs.py:422
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/transform_pairs.py:422: DeprecationWarning: invalid escape sequence \e
    """**profile6**:

abel/tools/transform_pairs.py:470
  /home/stg121/Software/Python/PyAbel/PyAbel-MR-tools.vmi/abel/tools/transform_pairs.py:470: DeprecationWarning: invalid escape sequence \e
    """**profile7**:

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================== 90 passed, 14 warnings in 17.51 seconds ====================
```

**Comment by MikhailRyazanov** [comment]
@stggh, thanks for the `toPES` suggestion! I've changed the test accordingly.

As for the `"""` warnings, they are in fact about TeX and Sphinx escapes that are not Python-escaped (like `'\sqrt'` instead of `'\\sqrt'` or `r'\sqrt'`). It's strange that they don't show up in the CI test, neither I saw them locally. However, pycodestyle indeed detects them, so I'll include the corrections in the next PR.

@DanHickstein, yes, these test are very basic and, in principle, something deeper is needed. I think, however, that most other stuff, like non-default and non-integer origin, or the resolution, should be done in tests of `polar.py`, which we in fact don't have at all... I'd be glad if somebody else can work on this (and other missing tests). Maybe, we should broaden #290 to list _all_ necessary test improvements and then, on occasion, start working on them?

**Pull request merged**